crates/ruma-state-res/src/room_version.rs

```diff
@@ -157,6 +177,61 @@ impl RoomVersion {
             strict_canonicaljson: true,
             limit_notifications_power_levels: true,
             extra_redaction_checks: true,
+            allow_knocking: false,
+            restricted_join_rules: false,
+        }
+    }
+
+    #[cfg(feature = "unstable-pre-spec")]
+    pub fn version_7() -> Self {
+        Self {
+            version: RoomVersionId::Version7,
+            disposition: RoomDisposition::Stable,
+            event_format: EventFormatVersion::V3,
+            state_res: StateResolutionVersion::V2,
+            enforce_key_validity: true,
+            special_case_aliases_auth: false,
+            strict_canonicaljson: true,
+            limit_notifications_power_levels: true,
+            extra_redaction_checks: true,
+            allow_knocking: true,
+            restricted_join_rules: false,
+        }
+    }
+
+    #[cfg(feature = "unstable-pre-spec")]
+    pub fn version_8() -> Self {
+        Self {
+            version: RoomVersionId::Version8,
+            disposition: RoomDisposition::Stable,
+            event_format: EventFormatVersion::V3,
+            state_res: StateResolutionVersion::V2,
+            enforce_key_validity: true,
+            special_case_aliases_auth: false,
+            strict_canonicaljson: true,
+            limit_notifications_power_levels: true,
+            extra_redaction_checks: true,
+            // Wait so should this be false?
+            // because: https://github.com/matrix-org/matrix-doc/pull/3289#issuecomment-884165177
+            allow_knocking: false,
```

Comment:
See the comment in the link https://github.com/matrix-org/matrix-doc/pull/3289#issuecomment-884165177

Reply:
To me it seems like the comment says the opposite, that this should be `true` so v8 is a simple extension of v7 rather than both being different extensions of v6.